core/cat/factory/custom_embedder.py

```diff
@@ -64,3 +64,23 @@ def embed_query(self, text: str) -> List[float]:
         ret = httpx.post(self.url, data=payload, timeout=None)
         ret.raise_for_status()
         return ret.json()["data"][0]["embedding"]
+
+
+class CustomOllamaEmbeddings(Embeddings):
+    """Use Ollama to serve embedding models."""
+
+    def __init__(self, base_url, model):
+        self.url = os.path.join(base_url, "api/embed")
+        self.model = model
+
+    def embed_documents(self, texts: List[str]) -> List[List[float]]:
+        payload = json.dumps({"model": self.model , "input": texts})
+        ret = httpx.post(self.url, data=payload, timeout=None)
+        ret.raise_for_status()
+        return ret.json()["embeddings"]
+
+    def embed_query(self, text: str) -> List[float]:
```

Comment:
@pieroit I implemented using `httpx` following the style of other customer embedders.

However, I am doubt if I should refactory with the `import ollama` module, to do more complex logic like downloading on the fly models:

examples: 

* https://github.com/ollama/ollama-python?tab=readme-ov-file#embed
* https://github.com/ollama/ollama-python?tab=readme-ov-file#errors

Reply:
This will do for now! Thanks @zioproto